### PR TITLE
[NETBEANS-3538]:Added Delete Action for Java Modular Project

### PR DESCRIPTION
**Pull request by Mohan-Sarilla**
**Jira Link :** 
https://issues.apache.org/jira/browse/NETBEANS-3538

**Description:**
Steps to simulate:

Create Java Modular Project:

Ctrl+shift+n > Dialogue box will be opened > from Categories select Java with Ant > from Projects select Java Modular Project > click Next > enter project name in Project Name box > click on Finish.
          After performing above steps Java Modular Project will be created.

Create Module under Java Modular Project:

Right click on created project > New > Module > Dialogue box will be opened > enter module name in Module Name field > click on Finish.
After performing above steps Module will be created.

then Try to right click on module that you created , you would see there will no Delete option.

### CODE REVIEW

File: java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java [hunk 1]
```diff
@@ -557,12 +561,40 @@ public String getHtmlDisplayName() {
                         null,
                         SystemAction.get(FileSystemAction.class ),
                         null,
-                        SystemAction.get(ToolsAction.class )
+                        SystemAction.get(ToolsAction.class ),
+                        //Added new action to delete module
+                        new DeleteAction("Delete") // NOI18N
```
Comment (matthiasblaesing) [inline java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: Please use  `org.openide.actions.DeleteAction`

File: java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java [hunk 2]
```diff
@@ -557,12 +561,40 @@ public String getHtmlDisplayName() {
                         null,
                         SystemAction.get(FileSystemAction.class ),
                         null,
-                        SystemAction.get(ToolsAction.class )
+                        SystemAction.get(ToolsAction.class ),
+                        //Added new action to delete module
```
Comment (matthiasblaesing) [inline java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: From the name of the action the intention is clear, new is relative (in a few years it is not new anymore) -> please remove the comment.

File: java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java [hunk 3]
```diff
@@ -557,12 +561,40 @@ public String getHtmlDisplayName() {
...
+                        new DeleteAction("Delete") // NOI18N
                     };
                 }
                 return actions;
             }
         }
+        
+        private class DeleteAction extends AbstractAction {
```
Comment (matthiasblaesing) [inline java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: Please don't reinvent the wheel. The explorer view already handles the confirmation - override `canDestroy` and `destroy`  in the `Node` implementation and you should be done.

**Comment by Mohan-Sarilla** [comment]
@matthiasblaesing , done with the changes requested.

File: java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java [hunk 4]
```diff
@@ -465,6 +466,100 @@ public boolean equals(@NullAllowed final Object other) {
                     new DynLkp());
         }
 
+        @Override
+        public boolean canDestroy() {
+            return true;
+        }
+
+        @Override
+        public void destroy() {
```
Comment (matthiasblaesing) [inline java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: Why did the delete routine grow from 20 lines (original commit to nearly 100 lines)?

Comment (Mohan-Sarilla) [reply java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: @matthiasblaesing, I haven't deregister classes, packages like it does in package deletion in previous commit, I have done the same in current commit.

Comment (matthiasblaesing) [reply java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java]: But is this all necessary? I can delete files outside the IDE and my experience with netbeans is, that it correctly detects this and updates the state of the projects.

**Comment by matthiasblaesing** [comment]
Ne reply to open questions.

**Pull request closed without merging**